src/Stratis.Bitcoin.Features.PoA/BasePoAFeatureConsensusRules/PoAHeaderSignatureRule.cs

```diff
@@ -33,6 +53,42 @@ public override void Run(RuleContext context)
 
             if (!this.validator.VerifySignature(pubKey, header))
             {
+                if (this.votingEnabled)
```

Comment:
Can voting not go in its own rule?
Then one can not just enable voting by changing a consensus field.
Also it keeps a clean separation between logical behaviours 

Also what's the purpose of max reorg exactly can you explain?

This is quite an important component more comments are required

Reply:
Added comments